File: edu/src/main/java/io/agora/edu/common/service/ChatService.java
package io.agora.edu.common.service;

import io.agora.edu.common.bean.ResponseBody;
import io.agora.edu.common.bean.request.ChatTranslateReq;
import io.agora.edu.common.bean.response.ChatRecordRes;
import io.agora.edu.common.bean.response.ChatTranslateRes;
import io.agora.edu.common.bean.response.SendChatRes;
import io.agora.education.impl.user.data.request.EduRoomChatMsgReq;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ChatService {
    @POST("edu/apps/{appId}/v2/rooms/{roomUuid}/from/{userUuid}/chat")
    Call<SendChatRes> roomChat(
            @Path("appId") String appId,
            @Path("roomUuid") String roomUuid,
            @Path("userUuid") String fromUuid,
            @Body EduRoomChatMsgReq req);

    @POST("edu/acadsoc/apps/{appId}/v1/translation")
    Call<ResponseBody<ChatTranslateRes>> translate(
            @Path("appId") String appId,
            @Body ChatTranslateReq req);

    @GET("edu/apps/{appId}/v2/rooms/{roomUUid}/chat/messages")
    Call<ResponseBody<ChatRecordRes>> pullChatRecords(
            @Path("appId") String appId,
            @Path("roomUUid") String roomUUid,
            @Query("count") int count,
            @Query("nextId") String nextId,
            @Query("sort") int sort);
}

File: edu/src/main/java/io/agora/edu/core/internal/server/requests/http/retrofit/services/deprecated/ChatService.kt
package io.agora.edu.core.internal.server.requests.http.retrofit.services.deprecated

import io.agora.edu.core.internal.edu.common.bean.ResponseBody
import io.agora.edu.core.internal.server.struct.request.*
import io.agora.edu.core.internal.server.struct.response.*
import retrofit2.Call
import retrofit2.http.*

interface ChatService {
    @POST("edu/apps/{appId}/v2/rooms/{roomUuid}/from/{userUuid}/chat")
    fun roomChat(
            @Path("appId") appId: String?,
            @Path("roomUuid") roomUuid: String?,
            @Path("userUuid") fromUuid: String?,
            @Body req: EduRoomChatMsgReq?): Call<SendChatRes>?

    @POST("edu/acadsoc/apps/{appId}/v1/translation")
    fun translate(
            @Path("appId") appId: String?,
            @Body req: ChatTranslateReq?): Call<DataResponseBody<ChatTranslateRes?>?>?

    @GET("edu/apps/{appId}/v2/rooms/{roomUUid}/chat/messages")
    fun pullChatRecords(
            @Path("appId") appId: String?,
            @Path("roomUUid") roomUUid: String?,
            @Query("count") count: Int,
            @Query("nextId") nextId: String?,
            @Query("sort") sort: Int): Call<DataResponseBody<ChatRecordRes>>?

    /**
     * Conversation is defined as only used in student Q&A session
     */
    @POST("edu/apps/{appId}/v2/rooms/{roomUuid}/conversation/students/{studentUuid}/messages")
    fun conversation(
            @Path("appId") appId: String?,
            @Path("roomUuid") roomUuid: String?,
            @Path("studentUuid") userUuid: String?,
            @Body req: EduRoomChatMsgReq?): Call<ConversationRes?>?

    @GET("edu/apps/{appId}/v2/rooms/{roomUUid}/conversation/students/{studentsUuid}/messages")
    fun pullConversationRecords(
            @Path("appId") appId: String?,
            @Path("roomUUid") roomUUid: String?,
            @Path("studentsUuid") userUuid: String?,
            @Query("nextId") nextId: String?,
            @Query("sort") sort: Int): Call<DataResponseBody<ConversationRecordRes>>?







    /**发送自定义的频道消息*/
    @POST("scene/apps/{appId}/v1/rooms/{roomUuid}/message/channel")
    fun sendChannelCustomMessage(
            @Path("appId") appId: String,
            @Path("roomUuid") roomUuid: String,
            @Body eduRoomMsgReq: EduRoomMsgReq
    ): Call<ResponseBody<String>>

    /**发送自定义的点对点消息*/
    @POST("scene/apps/{appId}/v1/rooms/{roomUuid}/users/{toUserUuid}/messages/peer")
    fun sendPeerCustomMessage(
            @Path("appId") appId: String,
            @Path("roomUuid") roomUuid: String,
            @Path("toUserUuid") toUserUuid: String,
            @Body eduUserMsgReq: EduUserMsgReq
    ): Call<ResponseBody<String>>



    /**发送用户间的私聊消息*/
    @POST("scene/apps/{appId}/v1/rooms/{roomUuid}/users/{toUserUuid}/chat/peer")
    fun sendPeerChatMsg(
            @Path("appId") appId: String,
            @Path("roomUuid") roomUuid: String,
            @Path("toUserUuid") toUserUuid: String,
            @Body eduUserChatMsgReq: EduUserChatMsgReq
    ): Call<ResponseBody<String>>
}